src/models/User.js

```diff
@@ -57,5 +58,13 @@ const UserSchema = new mongoose.Schema({
   creditCardInfo: creditCardSchema,
 });
 
+//before we save the document we hash password; this points to the user (this= user)
+UserSchema.pre('save', async function () {
+  console.log(this.modifiedPaths()); //to see which properties were modified
+   console.log(this.isModified('name')); // in console we get false if we are not modifying name, and true if we do
+  
+  if (!this.isModified('password')) return; //if the password is not modified then stop
+  });
```

Comment:
Is there meant to be an else branch here of what to do if password was modified (or does that change live in another branch)?

Reply:
we will add it later, i just was recycling code from the previous project🫣